identity-account-storage/src/storage/stronghold.rs

```diff
@@ -365,6 +413,55 @@ async fn sign_ed25519(vault: &Vault<'_>, payload: Vec<u8>, location: &KeyLocatio
   Ok(Signature::new(signature.into()))
 }
 
+async fn diffie_hellman(
+  vault: &Vault<'_>,
+  private_key: &KeyLocation,
+  public_key: [u8; X25519::PUBLIC_KEY_LENGTH],
+  shared_key: &KeyLocation,
+) -> Result<()> {
+  let diffie_hellman: procedures::X25519DiffieHellman = procedures::X25519DiffieHellman {
+    public_key,
+    private_key: private_key.into(),
+    shared_key: shared_key.into(),
+    hint: default_hint(),
+  };
+  vault.execute(diffie_hellman).await.map_err(Into::into)
+}
+
+async fn aead_encrypt(
+  vault: &Vault<'_>,
+  location: &KeyLocation,
+  plaintext: Vec<u8>,
+  associated_data: Vec<u8>,
+) -> Result<EncryptedData> {
+  let aead_encrypt: procedures::AeadEncrypt = procedures::AeadEncrypt {
+    cipher: procedures::AeadCipher::Aes256Gcm,
+    associated_data,
+    plaintext,
+    nonce: [0; Aes256Gcm::NONCE_LENGTH].to_vec(),
```

Comment:
What should we do here? Nonce has to be the same when encrypting and decrypting the data